### PR TITLE
Refactor Camera Control and Implement Accesibility

### PR DESCRIPTION
**Pull request by dgomezc**
## PR checklist

Quick summary of changes

- Which issue does this PR relate to?
#519

Finally we aren´t going to create a CameraService since the logic is very coupled to the user control.

Moving the camera logic that is on the pages (buttons and control initialization) to the camera control itself.

Created dependency properties for customize button styles.

Improved accessibility.

Ref: #908

### CODE REVIEW

File: templates/_composition/_shared/Page.Camera.AddResource/Controls/CameraControl.xaml.cs [hunk 1]
```diff
@@ -65,24 +72,107 @@ public bool IsInitialized
...
+                _mediaCapture.Failed -= MediaCapture_Failed;
+                _mediaCapture.Dispose();
+                _mediaCapture = null;
             }
         }
 
-        public async Task<string> TakePhotoAsync()
+        public async Task<string> TakePhoto()
         {
+            if (_capturing)
```
Comment (CarteKiwi) [inline templates/_composition/_shared/Page.Camera.AddResource/Controls/CameraControl.xaml.cs:169]: You are never setting _capturing back to false.

File: templates/_composition/_shared/Page.Camera.AddResource/Controls/CameraControl.xaml.cs [hunk 2]
```diff
@@ -94,98 +184,59 @@ public bool IsInitialized
                     photoOrientation = PhotoOrientation.FlipHorizontal;
                 }
 
-                return await ReencodeAndSavePhotoAsync(stream, photoOrientation);
+                var photo = await ReencodeAndSavePhotoAsync(stream, photoOrientation);
+                PhotoTaken?.Invoke(this, new CameraControlEventArgs(photo));
+                return photo;
```
Comment (CarteKiwi) [inline templates/_composition/_shared/Page.Camera.AddResource/Controls/CameraControl.xaml.cs]: Set _capturing to false

**Review by CarteKiwi — CHANGES_REQUESTED** [review]

File: templates/_composition/_shared/Page.Camera.AddResource/Controls/CameraControl.xaml.cs (hunk 1 above)
Comment (dgomezc) [reply templates/_composition/_shared/Page.Camera.AddResource/Controls/CameraControl.xaml.cs:169]: @CarteKiwi You're right, it's already fixed.

File: templates/_composition/_shared/Page.Camera.AddResource/Controls/CameraControl.xaml.cs (hunk 2 above)
Comment (dgomezc) [reply templates/_composition/_shared/Page.Camera.AddResource/Controls/CameraControl.xaml.cs]: @CarteKiwi You're right, it's already fixed.

**Review by CarteKiwi — APPROVED** [review]

**Pull request merged**